### PR TITLE
EZP-28840: As an editor I want to bookmark a content item from Content View

### PR DESCRIPTION
**Pull request by ViniTou**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-28840
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | yes
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

This needs _some_ css improvments and is based on current state of:
https://github.com/ezsystems/ezpublish-kernel/pull/2324

bookmark

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/bundle/Resources/public/js/scripts/admin.location.bookmark.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+(function (global, doc) {
+    const updateBookmark = event => {
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/admin.location.bookmark.js]: `(event) => {`

File: src/bundle/Resources/views/parts/form/location_bookmark.html.twig [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{% trans_default_domain 'locationview' %}
+
+<div>
+    {% if form.vars.value.bookmarked %}
+        <div title = {{ 'location.bookmark.remove'|trans|desc('Remove from Bookmarks') }}>
```
Comment (dew326) [inline src/bundle/Resources/views/parts/form/location_bookmark.html.twig]: What about this fancy change animation?

File: src/bundle/Resources/public/js/scripts/admin.location.bookmark.js (hunk 1 above)
Comment (sunpietro) [reply src/bundle/Resources/public/js/scripts/admin.location.bookmark.js]: Even better: 
```javascript
const SELECTOR_FORM = 'form[name="location_update_bookmark"]';
const updateBookmark = () => doc.querySelector(SELECTOR_FORM).submit();
```

File: src/bundle/Resources/views/parts/form/location_bookmark.html.twig [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{% trans_default_domain 'locationview' %}
+
+<div>
```
Comment (sunpietro) [inline src/bundle/Resources/views/parts/form/location_bookmark.html.twig]: What's the point of having a `div` without class or anything?

File: src/bundle/Resources/views/parts/form/location_bookmark.html.twig [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    {% if form.vars.value.bookmarked %}
+        <div title = {{ 'location.bookmark.remove'|trans|desc('Remove from Bookmarks') }}>
+            <svg class="ez-icon">
+                <use xlink:href="{{ asset('bundles/ezplatformadminui/img/ez-icons.svg') }}#bookmark-active"></use>
+            </svg>
+        </div>
+    {% else %}
+        <div title = {{ 'location.bookmark.add'|trans|desc('Add to Bookmarks') }}>
```
Comment (sunpietro) [inline src/bundle/Resources/views/parts/form/location_bookmark.html.twig]: Do you need a `div` here?

Comment (ViniTou) [reply src/bundle/Resources/views/parts/form/location_bookmark.html.twig]: Can't add title to svg tag I believe.

Comment (sunpietro) [reply src/bundle/Resources/views/parts/form/location_bookmark.html.twig]: Anyway, @dew326 told me the HTML structure will be rewritten to implement an animation.

**Comment by dew326** [comment]
@sunpietro @tischsoic frontend ready for review.

File: src/bundle/Resources/public/scss/_add-to-bookmarks.scss [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      position: relative;
+      border-radius: 50%;
+      cursor: pointer;
+
+      &::after {
+         display: block;
+         margin-left: -1px;
+         border-radius: 50%;
+         content: "";
+         animation: background-toTransparent .25s ease-in;
```
Comment (sunpietro) [inline src/bundle/Resources/public/scss/_add-to-bookmarks.scss]: `background-to-transparent`

File: src/bundle/Resources/public/scss/_add-to-bookmarks.scss [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+         height: 3rem;
+         padding: .4rem;
+         transition: all .25s ease-in;
+      }
+   }
+
+   &__checkbox:checked + &__label {
+      &::after {
+         background-color: #fff;
+         animation: background-toWhite .25s ease-in;
```
Comment (sunpietro) [inline src/bundle/Resources/public/scss/_add-to-bookmarks.scss]: `background-to-white`

File: src/bundle/Resources/views/content/locationview.html.twig [hunk 7]
```diff
@@ -132,6 +135,7 @@
         'bundles/ezplatformadminui/js/scripts/admin.location.add.custom_url.js'
         '@EzPlatformAdminUiBundle/Resources/public/js/scripts/button.state.toggle.js'
         '@EzPlatformAdminUiBundle/Resources/public/js/scripts/admin.version.edit.conflict.js'
+        'bundles/ezplatformadminui/js/scripts/admin.location.bookmark.js'
```
Comment (mikadamczyk) [inline src/bundle/Resources/views/content/locationview.html.twig]: Which form should we use? 'bundles/ezplatformadminui/js/scripts/admin.location.bookmark.js' or '@EzPlatformAdminUiBundle/Resources/public/js/scripts/admin.location.bookmark.js'?

Comment (webhdx) [reply src/bundle/Resources/views/content/locationview.html.twig]: @mikadamczyk @ViniTou I'd rather use `@EzPlatformAdminUiBundle` as we use resource names for twig templates.

**Comment by lserwatka** [comment]
@dew326 could you approve it?

**Pull request merged**